rewrite-gradle/src/main/java/org/openrewrite/gradle/trait/GradleDependency.java

```diff
@@ -0,0 +1,167 @@
+                if (configuration == null) {
+                    return null;
+                }
+
+                org.openrewrite.gradle.util.Dependency dependency = null;
+                Expression argument = methodInvocation.getArguments().get(0);
+                if (argument instanceof J.Literal || argument instanceof G.GString || argument instanceof G.MapEntry) {
+                    dependency = parseDependency(methodInvocation.getArguments());
+                } else if (argument instanceof J.MethodInvocation &&
+                        (((J.MethodInvocation) argument).getSimpleName().equals("platform") ||
+                                ((J.MethodInvocation) argument).getSimpleName().equals("enforcedPlatform"))) {
+                    dependency = parseDependency(((J.MethodInvocation) argument).getArguments());
+                }
+
+                if (dependency == null) {
+                    return null;
+                }
+
+                if (configuration.isCanBeResolved()) {
+                    for (ResolvedDependency resolvedDependency : configuration.getResolved()) {
+                        if ((groupId == null || matchesGlob(resolvedDependency.getGroupId(), groupId)) &&
+                                (artifactId == null || matchesGlob(resolvedDependency.getArtifactId(), artifactId))) {
+                            Dependency req = resolvedDependency.getRequested();
+                            if ((req.getGroupId() == null || req.getGroupId().equals(dependency.getGroupId())) &&
+                                    req.getArtifactId().equals(dependency.getArtifactId())) {
+                                return new GradleDependency(cursor, resolvedDependency);
+                            }
+                        }
+                    }
+                } else {
+                    for (GradleDependencyConfiguration transitiveConfiguration : gradleProject.configurationsExtendingFrom(configuration, true)) {
+                        if (transitiveConfiguration.isCanBeResolved()) {
+                            for (ResolvedDependency resolvedDependency : transitiveConfiguration.getResolved()) {
+                                if ((groupId == null || matchesGlob(resolvedDependency.getGroupId(), groupId)) &&
+                                        (artifactId == null || matchesGlob(resolvedDependency.getArtifactId(), artifactId))) {
+                                    Dependency req = resolvedDependency.getRequested();
+                                    if ((req.getGroupId() == null || req.getGroupId().equals(dependency.getGroupId())) &&
+                                            req.getArtifactId().equals(dependency.getArtifactId())) {
+                                        return new GradleDependency(cursor, resolvedDependency);
+                                    }
+                                }
+                            }
```

Comment:
callout: I wasn't quite sure what should happen here, so I opted to just return the first `ResolvedDependency` encountered.